.github/workflows/macos.yml

```diff
@@ -18,6 +25,28 @@ jobs:
       with:
         java-version: 11
         distribution: 'temurin'
+    - name: Build GDAL
+      run: |
+        mkdir $GDAL_INSTALL_PATH
+        brew update
+        brew upgrade
+        brew install freexl libxml2 libspatialite geos proj libgeotiff openjpeg libaec postgis poppler doxygen unixodbc aom jpeg-xl libheif libarchive libkml boost ccache swig python-setuptools
+        : # gdal is automatically installed as a dependency for postgis
+        brew uninstall --ignore-dependencies gdal
+        wget https://github.com/OSGeo/gdal/releases/download/v$OSX_GDAL_VERSION/gdal-$OSX_GDAL_VERSION.tar.gz
+        tar -xvf gdal-$OSX_GDAL_VERSION.tar.gz
+        cd gdal-$OSX_GDAL_VERSION
+        mkdir build
+        cd build
+        : # See https://github.com/OSGeo/gdal/issues/6949 (Java bindings) and https://github.com/OSGeo/gdal/pull/8226 (Arrow and Parquet)
+        cmake -DCMAKE_INSTALL_PREFIX=$GDAL_INSTALL_PATH -DCMAKE_BUILD_TYPE=Release -DBUILD_JAVA_BINDINGS=ON -DBUILD_PYTHON_BINDINGS=OFF ${CMAKE_OPTIONS} ..  
+        cmake --build .
+        cmake --build . --target install
+        ls /Users/runner/work/gdal/lib/jni/
+        ls /usr/lib/
+        otool -L $GDAL_INSTALL_PATH/lib/jni/libgdalalljni.dylib
```

Comment:
Seems to be because of [this](https://stackoverflow.com/questions/70549365/why-are-my-system-libraries-and-frameworks-not-visible-in-macos-monterey) - all the system libraries are now virtual:
> As part of this change, copies of dynamic libraries are no longer present on the filesystem. Code that attempts to check for dynamic library presence by looking for a file at a path or enumerating a directory will fail. Instead, check for library presence by attempting to dlopen() the path, which will correctly check for the library in the cache

No clue if that's a problem or not.